src/manager.cc

```diff
@@ -43,15 +43,15 @@ void event_remove_and_close(Event* event) { utils::ThreadInternal::poll()->remov
 } // namespace this_thread
 
 Manager::Manager()
-  : m_chunk_manager(new ChunkManager),
-    m_connection_manager(new ConnectionManager),
-    m_download_manager(new DownloadManager),
-    m_file_manager(new FileManager),
-    m_handshake_manager(new HandshakeManager),
-    m_resource_manager(new ResourceManager),
-
-    m_client_list(new ClientList),
-    m_dht_controller(new tracker::DhtController),
+  : m_chunk_manager(std::make_unique<ChunkManager>()),
+    m_connection_manager(std::make_unique<ConnectionManager>()),
+    m_download_manager(std::make_unique<DownloadManager>()),
+    m_file_manager(std::make_unique<FileManager>()),
+    m_handshake_manager(std::make_unique<HandshakeManager>()),
+    m_resource_manager(std::make_unique<ResourceManager>()),
+
+    m_client_list(std::make_unique<ClientList>()),
+    m_dht_controller(std::make_unique<tracker::DhtController>()),
```

Comment:
These don't actually do anything to clean up the code.

Reply:
The type is in the class declaration, and we can't mistakenly pass the wrong types to the ctor.

Difference from other cases is that here we are calling the unique_ptr ctor directly, while make_unique is a helper method to call unique_ptr ctor.

Reply:
Depends, for initializing member variables in the constructor as above, then yes.

However when you pass a unique_ptr to a constructor within a code block, then using make_unique actually tells the reader useful information.

As such the former it makes sense to make it less verbose, while the latter you want to write it out by using the _helper function_ make_unique.

Reply:
OK done